vision_stack/camera/bottomcam/test/test_cam_opens.cpp

```diff
@@ -0,0 +1,23 @@
+// Bring in my package's API, which is what I'm testing
+// #include "../src/bottomcam.cpp"
+// Bring in gtest
+#include <gtest/gtest.h>
+
+// Declare a test
+TEST(TestSuite, testCase1)
+{
+  EXPECT_EQ(1, 1); // 1 == 1
+  EXPECT_GT(10, 0); // 10 > 0
+}
+/*
+// Declare another test
+TEST(TestSuite, testCase2)
+{
+<test things here, calling EXPECT_* and/or ASSERT_* macros as needed>
+}
+*/
+// Run all the tests that were declared with TEST()
+int main(int argc, char **argv){
+  testing::InitGoogleTest(&argc, argv);
```

Comment:
@hargup Thanks for jumping in!

[Google Test](http://wiki.ros.org/rostest/Writing#With_gtest) is the CPP side of [rostest](http://wiki.ros.org/rostest).  
I am basically restructuring this package so that we can separate the API, and the `main()` function. [Previously, we had code](https://github.com/auviitkgp/kraken_3.0/blob/d7c7d1cdcc6deacb10f62c50df6a880b5c25db1b/vision_stack/camera/bottomcam/src/bottomcam.cpp#L1) that had the API and the `main()` function baked into the same file, and thus, when I tried to bring it in the test CPP file, it threw an error. (2 `main` functions.)  
You can see the [file `test/*.cpp`](https://github.com/icyflame/kraken_3.0/blob/b4dcd710512c49a59483e8179f89f36a381a1181/vision_stack/camera/bottomcam/test/test_cam_opens.cpp) which is the test file. I plan to include the `*.h` API there, and then, write tests for it. The tests work fine for now, but I have not written any meaningful tests yet.  
(Currently, there are only some sanity tests there. 1 == 1 and 10 > 0 :smile: )

***

cc @nevinvalsaraj @prudhvid